Makefile

```diff
@@ -0,0 +1,42 @@
+check: check-leaders check-workers check-first check-optimizer
+
+format: format-first format-optimizer
+
+check-leaders:
```

Comment:
I think check-workers and check-leaders should just call a PHONY command that runs `cd leaders && ./gradlew check`

Reply:
A little confused by how to do this in make. I updated the Makefile so that all of the commands are PHONY commands (as they should be since they don't depend on files) but I'm not entirely sure how to make a single command run after a specified target

Reply:
Like this:
```
# First command
command1:
	@cd dir1 && $(MAKE) -s do_something

# Second command
command2:
	@cd dir2 && $(MAKE) -s do_something

# Marking do_something as a phony target
.PHONY: do_something

# The .PHONY command to be run in each directory
do_something:
	@echo "Doing something in `pwd`"
```